src/internet_identity/src/storage/registration_rates.rs

```diff
@@ -281,6 +272,71 @@ mod test {
         );
     }
 
+    #[test]
+    fn previous_config_should_not_affect_new_config() {
+        let mut registration_rates = setup();
+        // initialize reference rate with 1 registration / second
+        for _ in 0..1000 {
+            registration_rates.new_registration();
+            TIME.with_borrow_mut(|t| *t += Duration::from_secs(1).as_nanos() as u64);
+        }
+
+        // raise current rate to 3 registrations / second
+        for _ in 0..100 {
+            registration_rates.new_registration();
+            registration_rates.new_registration();
+            registration_rates.new_registration();
+            TIME.with_borrow_mut(|t| *t += Duration::from_secs(1).as_nanos() as u64);
+        }
+
+        assert_eq!(
+            registration_rates.registration_rates().unwrap(),
+            NormalizedRegistrationRates {
+                current_rate_per_second: 3.0,
+                reference_rate_per_second: 1.2, // increases too, but more slowly
+                captcha_threshold_rate: 1.44,   // reference rate * 1.2 (as per config)
+            }
+        );
+
+        // Change the config.
+        state::persistent_state_mut(|ps| {
+            ps.captcha_config = CaptchaConfig {
+                max_unsolved_captchas: 500,
+                captcha_trigger: CaptchaTrigger::Dynamic {
+                    threshold_pct: 10,
+                    current_rate_sampling_interval_s: 100,
+                    reference_rate_sampling_interval_s: 100,
+                },
+            }
+        });
+
+        // Needed to prune the data after new config
+        // It prunes 100 for each new registration
+        // we added 1100 before. Remove them with 110, plus 2 to remove the new 110 added.
+        for _ in 0..112 {
+            registration_rates.new_registration();
+        }
+
+        // Pass enough time for the added registrations to also be pruned by new ones.
+        TIME.with_borrow_mut(|t| *t += Duration::from_secs(100).as_nanos() as u64);
+
+        // Set current rate to 2 registrations per seconds for 100 seconds
+        // which is the current and reference rate.
+        for _ in 0..100 {
+            registration_rates.new_registration();
+            registration_rates.new_registration();
+            TIME.with_borrow_mut(|t| *t += Duration::from_secs(1).as_nanos() as u64);
+        }
+        assert_eq!(
```

Comment:
Maybe change it to assert equality between `current_rate_per_second` and `reference_rate_per_second`? Because that is the important property of the test, not the actual resulting values.

Reply:
good point, done!